src/main/java/org/opensearch/index/store/directio/CryptoDirectIODirectory.java

```
@@ -196,7 +218,35 @@ public void deleteFile(String name) throws IOException {
                 LOGGER.warn("Failed to get file size", e);
             }
         }
-
+        EncryptionCache.getInstance().invalidateFile(filePath);
         super.deleteFile(name);
     }
+
+    /**
+     * Calculate content length with OSEF validation.
+     * Fast path: check cache first to avoid FileChannel open.
+     */
+    private long calculateContentLengthWithValidation(Path file, long rawFileSize) throws IOException {
+        if (rawFileSize < EncryptionMetadataTrailer.MIN_FOOTER_SIZE) {
+            return rawFileSize;
+        }
```

when would this even happen? should we rather throw an exception?